src/NativeScript/AllocatedPlaceholder.h

```diff
@@ -0,0 +1,48 @@
+//
+//  AllocatedPlaceholder.h
+//  NativeScript
+//
+//  Created by Ivan Buhov on 7/9/15.
+//
+//
+
+#ifndef __NativeScript__AllocatedPlaceholder__
+#define __NativeScript__AllocatedPlaceholder__
+
+namespace NativeScript {
+class AllocatedPlaceholder : public JSC::JSDestructibleObject {
+public:
+    typedef JSC::JSDestructibleObject Base;
+
+    static AllocatedPlaceholder* create(JSC::VM& vm, GlobalObject* globalObject, JSC::Structure* structure, id wrappedObject, JSC::Structure* instanceStructure) {
+        AllocatedPlaceholder* object = new (NotNull, JSC::allocateCell<AllocatedPlaceholder>(vm.heap)) AllocatedPlaceholder(vm, structure);
+        object->finishCreation(vm, globalObject, wrappedObject, instanceStructure);
+        return object;
+    }
+
+    DECLARE_INFO;
+
+    static JSC::Structure* createStructure(JSC::VM& vm, JSC::JSGlobalObject* globalObject, JSC::JSValue prototype) {
+        return JSC::Structure::create(vm, globalObject, prototype, JSC::TypeInfo(JSC::ObjectType, StructureFlags), info());
+    }
+
+    id wrappedObject() const {
+        return this->_wrappedObject;
+    }
+
+private:
+    id _wrappedObject;
+
+    AllocatedPlaceholder(JSC::VM& vm, JSC::Structure* structure)
+        : Base(vm, structure) {
+    }
+
+    void finishCreation(JSC::VM& vm, GlobalObject* globalObject, id wrappedObject, JSC::Structure* instanceStructure) {
+        Base::finishCreation(vm);
+        this->_wrappedObject = wrappedObject;
+        this->putDirect(vm, globalObject->instanceStructureIdentifier(), JSC::JSValue(instanceStructure), JSC::ReadOnly | JSC::DontEnum | JSC::DontDelete);
```

Comment:
You don't have to wrap `JSCell*` values in a `JSValue` call, because `JSValue` already has an implicit constructor that accepts `JSCell`. It's purely a cosmetic nitpick, but the rest of the codebase does not explicitly construct `JSValue` this way, so I suggest removing the explicit calls to the `JSValue` constructor in this patch.